### PR TITLE
C++ build fixes

### PR DESCRIPTION
**Pull request by externl**
This PR is the first step in getting the demos working again. Right now it just fixes the C++ builds. There's still optimizations to be made (using string_view, reducing optional usage, etc) in follow-up PRs.

### CODE REVIEW

File: cpp/Chat/client/Client.cpp [hunk 1]
```diff
@@ -39,132 +43,132 @@ class ChatRoomCallbackI : public Chat::ChatRoomCallback
...
+    optional<Glacier2::RouterPrx> router{communicator->getDefaultRouter()};
+    if (!router)
     {
-        if (argc > 1)
-        {
-            const string message = "usage: ";
-            throw runtime_error(message + appName());
-        }
+        cerr << "No router configured" << endl;
+        exit(1);
```
Comment (pepone) [inline cpp/Chat/client/Client.cpp:120]: replace with runtime_exception?

File: cpp/Chat/client/Client.cpp [hunk 2]
```diff
@@ -173,23 +177,44 @@ main(int argc, char* argv[])
     Ice::registerIceSSL();
```
[the remark this answers is not in the document]
Comment (externl) [reply cpp/Chat/client/Client.cpp:175]: I can do this everywhere right? We still need the rest?
```
Ice::registerIceUDP();
Ice::registerIceWS();
```

**Review by pepone — COMMENTED** [review]
A few comments, I will finish reviewing in a bit.

**Review by pepone — APPROVED** [review]

File: cpp/Manual/lifecycle/Client.cpp
[the remark this answers is not in the document]
Comment (externl) [reply cpp/Manual/lifecycle/Client.cpp:1]: We've already remove it from the docs.

File: cpp/IceStorm/replicated/application.xml
[the remark this answers is not in the document]
Comment (externl) [reply cpp/IceStorm/replicated/application.xml:1]: I updated the IceStorm service plugin version.

File: cpp/Chat/client/Client.cpp [hunk 3]
```diff
@@ -128,7 +128,7 @@ run(shared_ptr<Ice::Communicator> communicator)
     // provided by the router
     Ice::Identity id{Ice::generateUUID(), router->getCategoryForClient()};
 
-    Chat::ChatRoomCallbackPrx callbackPrx = adapter->add(make_shared<ChatRoomCallbackI>(), id);
+    auto callbackPrx = Chat::ChatRoomCallbackPrx{adapter->add(make_shared<ChatRoomCallbackI>(), id)};
```
Comment (pepone) [inline cpp/Chat/client/Client.cpp:131]: or 

```
Chat::ChatRoomCallbackPrx callbackPrx{adapter->add(make_shared<ChatRoomCallbackI>(), id)};
```

**Pull request merged**